ArduCopter/mode_rtl.cpp

```diff
@@ -358,20 +353,18 @@ bool Copter::ModeRTL::landing_gear_should_be_deployed() const
 //      called by rtl_run at 100hz or more
 void Copter::ModeRTL::land_run(bool disarm_on_land)
 {
-    // if not auto armed or landing completed or motor interlock not enabled set throttle to zero and exit immediately
-    if (!motors->armed() || !ap.auto_armed || ap.land_complete || !motors->get_interlock()) {
-        zero_throttle_and_relax_ac();
-        // set target to current position
-        loiter_nav->clear_pilot_desired_acceleration();
-        loiter_nav->init_target();
+    // check if we've completed this stage of RTL
+    _state_complete = ap.land_complete;
 
-        // disarm when the landing detector says we've landed
-        if (ap.land_complete && disarm_on_land) {
-            copter.init_disarm_motors();
-        }
+    // disarm when the landing detector says we've landed
+    if (disarm_on_land && ap.land_complete) {
```

Comment:
So just taking mode_rtl as an example.  Since init_disarm_motors is called from here, then in order to let it run make_safe_shut_down then we have to add a condition to this statement that requires the spool mode to be Ground Idle before allowing disarm.  Is this the way forward? if so, I can go through and make the changes to have it disarm properly in these cases.